src/idl_gen_general.cpp

```diff
@@ -656,20 +661,37 @@ class GeneralGenerator : public BaseGenerator {
   // Recursively generate arguments for a constructor, to deal with nested
   // structs.
   void GenStructArgs(const StructDef &struct_def, std::string *code_ptr,
-                     const char *nameprefix) const {
+                     const char *nameprefix, size_t array_count = 0) const {
     std::string &code = *code_ptr;
     for (auto it = struct_def.fields.vec.begin();
          it != struct_def.fields.vec.end(); ++it) {
       auto &field = **it;
-      if (IsStruct(field.value.type)) {
+      const auto field_type = field.value.type;
+      const auto array_field = IsArray(field_type);
+      const auto type = array_field ? field_type.VectorType()
```

Comment:
`const auto &type = field_type.VectorType()` is valid C++ construction.
The lifetime of returned temporary will be prolonged as long as it's a const reference.
[GotW 88](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/)

Better to use `const auto &` for VectorType() or remove it everywhere (idl_parser.cpp:1283, for example).
MSVC detects violation if `const` removed and generate a warning "A non-const reference may only be bound to an lvalue".

Reply:
Added.